lib/possible_bipartition.rb

```diff
@@ -1,4 +1,28 @@
 
 def possible_bipartition(dislikes)
-  raise NotImplementedError, "possible_bipartition isn't implemented yet"
+  visited = {}
+dislikes.length.times do |pup|
+  q = []
+  if !visited[pup]
+    q << [pup, "A"]
+  end
+
+  while !q.empty?
+    current_pup, current_group = q.shift
```

Comment:
Just a note that `q.shift` is an O(n) operation making this algorithm O(n^2 * E)  Where n is the number of nodes and E the number of edges.